apps/root/src/pages/earn/components/all-strategies-table/components/filters/index.tsx

```diff
@@ -0,0 +1,325 @@
+          {filteredOptions.length === 0 && (
+            <Typography variant="bodySmallBold">
+              <FormattedMessage defaultMessage="All" description="all-strategies-table.filters.all" />
+            </Typography>
+          )}
+        </ContainerBox>
+      </AccordionSummary>
+      <AccordionDetails>
+        {!hideSearch && (
+          <TextField
+            placeholder={intl.formatMessage(
+              defineMessage({
+                defaultMessage: 'Search',
+                description: 'all-strategies-table.filters.search',
+              })
+            )}
+            value={search}
+            onChange={(evt: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) =>
+              setSearch(evt.currentTarget.value)
+            }
+            autoFocus
+            InputProps={{
+              startAdornment: (
+                <InputAdornment position="start">
+                  <SearchIcon />
+                </InputAdornment>
+              ),
+            }}
+            onKeyDown={(e) => {
+              if (e.key !== 'Escape') {
+                // Prevents autoselecting item while typing (default Select behaviour)
+                e.stopPropagation();
+              }
+            }}
+          />
+        )}
+        <DividerBorder2 />
+        {optionsFilteredBySearch.length > 0 ? (
+          optionsFilteredBySearch.map((filter, index) => (
+            <FormGroup key={index}>
+              <FormControlLabel
+                control={
+                  <Checkbox
+                    onChange={() => onChangeFilter(filter.value)}
+                    checked={filteredOptions.includes(filter.value)}
+                  />
+                }
+                label={filter.label}
+              />
+            </FormGroup>
+          ))
+        ) : (
+          <Typography variant="bodySmallBold">
+            <FormattedMessage defaultMessage="No options found" description="all-strategies-table.filters.no-options" />
+          </Typography>
+        )}
+      </AccordionDetails>
+    </Accordion>
+  );
+};
+
+type FilterTypes = StrategyYieldType | string | number | Token;
+
+const TableFilters = ({ isLoading }: { isLoading: boolean }) => {
+  const [anchorEl, setAnchorEl] = React.useState<HTMLButtonElement | null>(null);
+  const dispatch = useAppDispatch();
+  const intl = useIntl();
+  const strategiesFilters = useAllStrategiesFilters();
+  const strategiesParameters = useStrategiesParameters();
+  const [expandedFilter, setExpandedFilter] = React.useState(0);
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const handleOpen = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const open = Boolean(anchorEl);
+  const id = open ? 'strategies-filters-popover' : undefined;
+
+  const filterItems = React.useMemo<FilterControl<FilterTypes>[]>(() => {
+    const assetFilter: FilterControl<Token> = {
```

Comment:
seems like a lot of this are just duplicated code, can we maybe make a function to generate those?

Reply:
I have created a helper function for this, but I believe some repetition is necessary due to the specific requirements of each option. Let me know if you have any other approaches in mind!